src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -64,15 +68,29 @@ public static EmptyInjectionConfiguration build(final int position, final boolea
     private int eventSize;
 
     public BatchItem(
-            final String event,
+            final String rawEvent,
             final boolean strictParsing,
             final EmptyInjectionConfiguration emptyInjectionConfiguration,
             final InjectionConfiguration[] injections,
             final List<Integer> skipCharacters) {
-        this.rawEvent = event;
+        this.rawEvent = rawEvent;
         this.skipCharacters = skipCharacters;
-        this.event = strictParsing ? StrictJsonParser.parseObject(event) : new JSONObject(event);
-        this.eventSize = event.getBytes(StandardCharsets.UTF_8).length;
+        this.event = new JSONObject(rawEvent);
+
+        // We will first release in a shadow mode when the strict parser does second parsing
+        // and we compare if the result is the same as with the old parser; we also log invalid events;
+        if (strictParsing) {
+            try {
+                final JSONObject shadowEvent = StrictJsonParser.parseObject(rawEvent);
+                if (!shadowEvent.toString().equals(this.event.toString())) {
```

Comment:
If the order of the fields in object will be the same here?

Reply:
I used the same approach as you used in unit test. I thought that it is safe :)

Reply:
I mean JSONObject doesn't really have equals method. So I can't compare them in a simple way.

Reply:
Then I agree with Ricardo that we don't need shadow parsing

Reply:
Then I agree with Ricardo that we don't need shadow parsing

Reply:
why?